src/entry/mod.rs

```diff
@@ -138,3 +139,24 @@ pub(crate) struct ZipEntryMeta {
     pub(crate) general_purpose_flag: GeneralPurposeFlag,
     pub(crate) file_offset: u64,
 }
+
+/// Stores information about a Zip entry inside of an archive. Besides storing archive independant
+/// information like the size and timestamp it can also be used to query information about how the
+/// entry is stored in an archive.
+#[derive(Clone)]
+pub struct StoredZipEntry {
+    pub entry: ZipEntry,
+    pub(crate) meta: ZipEntryMeta,
+}
+
+impl StoredZipEntry {
+    /// Returns the offset in bytes from where the data of the entry starts.
+    pub fn data_offset(&self) -> u64 {
+        let header_length = SIGNATURE_LENGTH + LFH_LENGTH;
+        let trailing_length = self.entry.comment().as_bytes().len()
```

Comment:
`trailing_length` here should actually only include the filename length and extra field length.

It looks like the original function was incorrect so that's my bad!